src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/importcontrol/InputImportControlCustomDtd.xml

```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<!DOCTYPE import-control PUBLIC
+        "CUSTOM_DTD"
+        "src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/importcontrol/InputImportControlCustom.dtd">
+
+<import-control pkg="com.puppycrawl.tools.checkstyle.checks.imports">
+  <allow class="java.awt.Image"/>
+  <subpackage name="importcontrol">
+    <someElementNotSupported class="java.awt.Image" local-only="true"/>
```

Comment:
We should throw exception for an unsupported element. Please open an issue more discussion on this.

Reply:
@nick-mancuso It seems a DTD is required, but I don't believe we check that a recognizable DTD is provided. Hence why we could provide some test DTD in this case.

Reply:
> @nick-mancuso It seems a DTD is required, but I don't believe we check that a recognizable DTD is provided. Hence why we could provide some test DTD in this case.

Exactly.

Reply:
Shouldn't the behavior of import configuration loader be the same as regular configuration loader?

```
➜  java git:(master) ✗ diff configuration_1_3.dtd configuration_custom.dtd 
8c8
< <!ELEMENT module (module|property|metadata|message)*>
---
> <!ELEMENT module (module|property|metadata|message|newelement)*>
52a53,58
>     value CDATA #REQUIRED
> >
> 
> <!ELEMENT newelement EMPTY>
> <!ATTLIST newelement
>     name NMTOKEN #REQUIRED


➜  java git:(master) ✗ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "CUSTOM_DTD" "configuration_1_4.dtd" >

<module name="Checker">
    <property name="fileExtensions" value="java,js,css,xml"/>
    <module name="LineLength">
        <property name="max" value="100"/>
        <newelement name="stuff_here" value="value"/>

    </module>
</module>


➜  java git:(master) ✗ java -jar checkstyle-10.3.1-all.jar -c config.xml Test.java
Exception in thread "main" java.lang.IllegalStateException: Unknown name:newelement.
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader.startElement(ConfigurationLoader.java:552)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:510)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractXMLDocumentParser.emptyElement(AbstractXMLDocumentParser.java:183)
        at java.xml/com.sun.org.apache.xerces.internal.impl.dtd.XMLDTDValidator.emptyElement(XMLDTDValidator.java:752)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanStartElement(XMLDocumentFragmentScannerImpl.java:1377)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2710)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:534)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:888)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:824)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1216)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:635)
        at com.puppycrawl.tools.checkstyle.XmlLoader.parseInputSource(XmlLoader.java:79)
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader.parseInputSource(ConfigurationLoader.java:199)
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:317)
        at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:268)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:372)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)

```

See here:
https://github.com/checkstyle/checkstyle/blob/dc74f7fa7b20d265146c2945a28851e5ae6db761/src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java#L549-L552

Reply:
That would be probably better, created #11983

Reply:
Changes done based on discussion done in #11983, https://github.com/checkstyle/checkstyle/issues/11983#issuecomment-1194353091